core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -2045,6 +2045,34 @@ class LogCleanerTest extends Logging {
     }
   }
 
+  @Test
+  def testMaxOverCleanerThreads(): Unit = {
+    val logCleaner =  new LogCleaner(new CleanerConfig(true),
+      logDirs = Array(TestUtils.tempDir(), TestUtils.tempDir()),
+      logs = new Pool[TopicPartition, UnifiedLog](),
+      logDirFailureChannel = new LogDirFailureChannel(1),
+      time = time)
+
+    val cleaners = mutable.ArrayBuffer[logCleaner.CleanerThread]()
+
+    val cleaner1 = new logCleaner.CleanerThread(1)
+    cleaner1.lastStats = new CleanerStats(time)
+    cleaner1.lastStats.bufferUtilization = 0.75d
+    cleaners += cleaner1
+
+    val cleaner2 = new logCleaner.CleanerThread(2)
+    cleaner2.lastStats = new CleanerStats(time)
+    cleaner2.lastStats.bufferUtilization = 0.85d
+    cleaners += cleaner2
+
+    val cleaner3 = new logCleaner.CleanerThread(3)
+    cleaner3.lastStats = new CleanerStats(time)
+    cleaner3.lastStats.bufferUtilization = 0.65d
+    cleaners += cleaner3
+
+    assertEquals(0, logCleaner.maxOverCleanerThreads(_.lastStats.bufferUtilization))
```

Comment:
Hmm, can we also assert a non zero value?

Reply:
Sure!
Just add it

Reply:
In reviewing this test, I noticed that I introduced a bug in #8783 that I round down the double value when updating metrics `max-buffer-utilization-percent`

I open https://issues.apache.org/jira/browse/KAFKA-18597 to fix it

Reply:
Good catch. I'll backport this one to 4.0 to make it easier to backport the bug fix.

Reply:
Also, given the fix, perhaps we don't need the case where we set the bufferUtilization to larger numbers.

Reply:
> Also, given the fix, perhaps we don't need the case where we set the bufferUtilization to larger numbers.

you are right. I copy your comment to #18627